packages/components/src/dropdown/types.ts

```diff
@@ -6,3 +23,86 @@ export type DropdownContentWrapperProps = {
 	 */
 	paddingSize?: 'none' | 'small' | 'medium';
 };
+
+export type DropdownProps = {
+	/**
+	 * If you want to target the dropdown menu for styling purposes,
+	 * you need to provide a contentClassName because it's not being rendered
+	 * as a child of the container node.
+	 */
+	contentClassName?: string;
+	/**
+	 * Opt-in prop to show popovers fullscreen on mobile.
+	 *
+	 * @default false
+	 */
+	expandOnMobile?: boolean;
+	/**
+	 * By default, the first tabbable element in the popover will receive focus
+	 * when it mounts. This is the same as setting this prop to "firstElement".
+	 * Specifying a true value will focus the container instead.
+	 * Specifying a false value disables the focus handling entirely
+	 * (this should only be done when an appropriately accessible
+	 * substitute behavior exists).
+	 *
+	 * @default 'firstElement'
+	 */
+	focusOnMount?: 'firstElement' | boolean;
+	/**
+	 * Set this to customize the text that is shown in the dropdown's header
+	 * when it is fullscreen on mobile.
+	 */
+	headerTitle?: string;
+	/**
+	 * A callback invoked when the popover should be closed.
+	 */
+	onClose?: () => void;
+	/**
+	 * A callback invoked when the state of the popover changes
+	 * from open to closed and vice versa.
+	 * The callback receives a boolean as a parameter.
+	 * If true, the popover will open.
+	 * If false, the popover will close.
+	 */
+	onToggle?: ( willOpen: boolean ) => void;
+	/**
+	 * Properties of popoverProps object will be passed as props
+	 * to the Popover component.
+	 * Use this object to access properties/features
+	 * of the Popover component that are not already exposed
+	 * in the Dropdown component,
+	 * e.g.: the ability to have the popover without an arrow.
+	 */
+	popoverProps?: WordPressComponentProps<
+		Omit< PopoverProps, 'children' >,
+		'div',
+		false
+	>;
```

Comment:
For this one, I'm thinking it would be better to do it this way:

```suggestion
	popoverProps?: Omit<
		ComponentPropsWithoutRef< typeof Popover >,
		'children'
	>;
```

The reason is so we can avoid duplicating the `WordPressComponentProps< ..., 'div', false >` part of it which is defined in Popover.

Reply:
Good idea to do `typeof Popover`, much more DRY.

Committed in https://github.com/WordPress/gutenberg/compare/33c67154869b08a170d6d78f8c4a55b2e3ed5b78...e693603136d8ebd61f2ab23b4b5bfe6c100d0ddc